### PR TITLE
docs: Update Caddy example

### PR DESCRIPTION
**Pull request by francislavoie**
Caddy v2 was released recently. This updates the https://router.vuejs.org/guide/essentials/history-mode.html#example-server-configurations page with an example that will work for Caddy v2, and adds the `v1` suffix for the old example.

### CODE REVIEW

File: docs/guide/essentials/history-mode.md
Comment (Alanscut) [inline docs/guide/essentials/history-mode.md:102]:
```Caddyflie
```Caddyfile

File: docs/guide/essentials/history-mode.md
Comment (Alanscut) [inline docs/guide/essentials/history-mode.md:108]:
```Caddyflie
```Caddyfile

**Review by Alanscut — COMMENTED** [review]
LGTM. But，it maybe much better to add a 'Caddyfile' here .

**Comment by francislavoie** [comment]
What syntax highlighter is used by these docs? The only ones I'm aware of that support Caddyfile syntax are Chroma (Golang one) which I implemented myself, and the Caddy VSCode plugin which uses textmate syntax.

**Comment by Alanscut** [comment]
Makes users understand the caddyfile syntax easier.

**Comment by francislavoie** [comment]
I think you misunderstood me. I'm not aware of a "Caddyfile" syntax highlighter other than the two I listed, so adding "Caddyfile" to this code block will likely do nothing. It'll still just render as regular text.

**Comment by Megrax** [comment]
@Alanscut @posva Is there anyone to review or approve this pull request ?

**Comment by francislavoie** [comment]
Well FWIW I'm one of the maintainers of Caddy, so IMO someone who is experienced with Caddy is not necessary here to review. You can take my word for it if lack of time is the issue. But I'd be glad to answer any questions you may have 👍

**Comment by Megrax** [comment]
I'm still a beginner who depend a lot on official documents, so I went into trouble following the official example. But I've solved the problem already. Just hope others would do it right. Thanks a lot anyway. 😆

**Review by posva — APPROVED** [review]
@francislavoie Sorry, I didn't know you were a maintainer of Caddy. I was waiting for a second review but that won't be necessary. Thanks!

**Pull request merged**
